### PR TITLE
feat(rpc-testing-utils): implement trace_call streaming

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
TraceCallStream type enable async streaming of trace_call results

TraceCallResult type to encapsulate success and error outcome from trace call

### CODE REVIEW

**Comment by DoTheBestToGetTheBest** [comment]
> @DoTheBestToGetTheBest please, for each PR write a clear and concise description that includes motivation for the changes and a brief overview of them. Also, some of your previous PRs need attention (#5267, #5321, #5329 etc). make sure to attend to them before opening new ones

oh my god, I'm so sorry I have to be such a bother. I didn't know about the descriptions, I thought a title was enough. Sorry again my friend, I'll be careful next time. Don't worry about my other PRs, at the moment I have exams that I am taking, I try to take some time to reth during my exams, as soon as I finish my exams, I will go back to my PRs. If my PRs are disturbing, I can close them, please let me know. Again i'm so sorry

**Review by mattsse — CHANGES_REQUESTED** [review]
nit

File: crates/rpc/rpc-testing-util/src/trace.rs [hunk 1]
```diff
@@ -97,6 +110,34 @@ pub trait TraceApiExt {
...
     where
         I: IntoIterator<Item = TraceFilter>;
+    /// Returns a new stream that yields the trace results for the given call requests.
+    fn trace_call_stream(
+        &self,
+        call: CallRequest,
+        trace_types: HashSet<TraceType>,
+        block_id: Option<BlockId>,
+        state_overrides: Option<StateOverride>,
+        block_overrides: Option<Box<BlockOverrides>>,
```
Comment (mattsse) [inline crates/rpc/rpc-testing-util/src/trace.rs:120]: this should be replaced with the new

https://github.com/paradigmxyz/reth/blob/e056bfb213e7ddb2f4345d9717735475229024dd/crates/rpc/rpc-types/src/eth/trace/tracerequest.rs#L12-L12

type which makes this a lot easier

**Review by mattsse — APPROVED** [review]
nice

**Pull request merged**